src/renderer/widgets/PasswordInputField.cpp

```diff
@@ -309,41 +305,55 @@ bool CPasswordInputField::draw(const SRenderData& data) {
     return dots.currentAmount != passwordLength || fade.animated || col.animated || redrawShadow || data.opacity < 1.0 || forceReload;
 }
 
-void CPasswordInputField::updateFailTex() {
-    const auto FAIL = g_pHyprlock->passwordLastFailReason();
-
-    if (checkWaiting)
-        placeholder.canGetNewFail = true;
-
+void CPasswordInputField::updatePlaceholder() {
     if (passwordLength != 0 || (checkWaiting && passwordLength == 0)) {
-        if (placeholder.failAsset) {
-            g_pRenderer->asyncResourceGatherer->unloadAsset(placeholder.failAsset);
-            placeholder.failAsset = nullptr;
-            placeholder.failID    = "";
-            redrawShadow          = true;
+        if (placeholder.asset && /* keep prompt asset cause it is likely to be used again */ placeholder.isFailText) {
+            std::erase(placeholder.registeredResourceIDs, placeholder.resourceID);
+            g_pRenderer->asyncResourceGatherer->unloadAsset(placeholder.asset);
+            placeholder.asset      = nullptr;
+            placeholder.resourceID = "";
+            redrawShadow           = true;
         }
         return;
     }
 
-    if (!FAIL.has_value() || !placeholder.canGetNewFail)
+    const auto AUTHFEEDBACKOPT = g_pAuth->getFeedback();
+    if (!AUTHFEEDBACKOPT.has_value())
+        return;
+
+    const auto AUTHFEEDBACK = AUTHFEEDBACKOPT.value();
+    if (AUTHFEEDBACK.text.empty() || placeholder.lastAuthFeedback == AUTHFEEDBACK.text)
+        return;
+
+    placeholder.isFailText       = AUTHFEEDBACK.isFail;
+    placeholder.lastAuthFeedback = AUTHFEEDBACK.text;
+
+    placeholder.asset = nullptr;
```

Comment:
TODO: I guess we want to keep the old asset around, so that we always have one to render